src/lib/optimizer/strategy/index_scan_rule.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
+
+/**
+ * This optimizer rule finds PredicateNodes whose children are StoredTableNodes. These PredicateNodes are candidates
+ * for being executed by IndexScans. If the expected selectivity of the predicate falls below a certain threshold the
+ * ScanType of the PredicateNode is set to IndexScan.
+ *
+ * Note:
+ * For now this rule is only applicable to single-column indexes and multi-column predicates (i.e. WHERE a < b) are
+ * not supported. We also assume that if chunks have an index, all of them are of the same type, we do not mix GroupKey
+ * and ART indexes. In addition, chains of IndexScans are not possible since an IndexScan's child must be a GetTable.
```

Comment:
Where do I find the GroupKey / ART thing in the code?

Reply:
The implementation of the data structures?

Reply:
No. Which part of your rule makes it so that you cannot mix them?

Reply:
The code simply only looks for GroupKey indexes

Reply:
Ah, I missed that line. Sorry.